### PR TITLE
Feat #15/create user service

### PR DESCRIPTION
**Pull request by ghost**
Changelog
feat(UserService): Create and implement core logic

Added user CRUD operations.
Implemented token generation and response handling.
fix(TokenInvalidException): Add single-argument constructor

Allows instantiation of the exception using only a message, improving flexibility.
feat(TokenResponseDTO): Add token field

Includes the token in the response to enhance API usability.
feat(Token): Add isTokenExpired and isTokenRevoked methods

Added methods for validating token expiration and revocation status, enabling more robust security checks.
Purpose
These changes collectively improve the service's functionality, code clarity, and maintainability while addressing necessary fixes for token handling and validation.

### CODE REVIEW

File: src/main/java/org/growify/bank/service/strategy/UserService.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
+package org.growify.bank.service.strategy;
+
+import org.growify.bank.dto.request.UptadeRequestDTO;
```
Comment (avila-r) [inline src/main/java/org/growify/bank/service/strategy/UserService.java:3]: "Uptade"?

Comment (avila-r) [inline src/main/java/org/growify/bank/service/strategy/UserService.java:3]: "Uptade"? Actual target is `dto.request.UpdateRequestDTO`

File: src/main/java/org/growify/bank/model/token/Token.java [hunk 2]
```diff
@@ -30,4 +30,7 @@ public class Token {
 
     @ManyToOne
     private User user;
-}
+
+    public boolean isTokenExpired() {
```
Comment (avila-r) [inline src/main/java/org/growify/bank/model/token/Token.java:34]: No reason for an explicit getter. Field may have `@lombok.Getter` (which is already implicitly defined by class annotation)

File: src/main/java/org/growify/bank/service/strategy/UserService.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+public class UserService {
+
+    private final TokenRepository tokenRepository;
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+    private final AuthenticationTokenManagerStrategy authTokenManager;
+    private final AuthenticateValidationStrategy authValidationStrategy;
+    private final PasswordValidationStrategy passwordValidationStrategy;
+    private final UserIdValidationStrategy userIdValidationStrategy;
+    private final EmailAlreadyValidationStrategy emailAlreadyValidationStrategy;
```
Comment (avila-r) [inline src/main/java/org/growify/bank/service/strategy/UserService.java:34]: Strategies must be obviously, explicitly imported:

image

File: src/main/java/org/growify/bank/service/strategy/UserService.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    private final ModelMapper modelMapper;
```
Comment (avila-r) [inline src/main/java/org/growify/bank/service/strategy/UserService.java:35]: There's no Model Mapper configured in the application context. A @Configuration class that publishes a `ModelMapper` bean is required to ensure dependency injection. An issue must be created to proceed.

File: src/main/java/org/growify/bank/service/strategy/UserService.java [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+    public ResponseEntity<List<UserResponseDTO>> getAllUsers() {
```
Comment (avila-r) [inline src/main/java/org/growify/bank/service/strategy/UserService.java:37]: Defining `ResponseEntity` or DTO operations in service-layer methods is a bad practice. Service classes should focus solely on business logic.

**Review by avila-r — CHANGES_REQUESTED** [review]

**Comment by avila-r** [comment]
Ensure the instructions are followed, then proceed with a new PR.

File: src/main/java/org/growify/bank/service/strategy/UserService.java (hunk 1 above)
Comment (ghost) [reply src/main/java/org/growify/bank/service/strategy/UserService.java:3]: You're right. I'll rename the DTO to UpdateRequestDTO and update all references in the code. Thank you for pointing that out!

File: src/main/java/org/growify/bank/model/token/Token.java (hunk 2 above)
Comment (ghost) [reply src/main/java/org/growify/bank/model/token/Token.java:34]: Understood. I’ll remove the explicit getters to simplify and follow the standard already used in the class.

File: src/main/java/org/growify/bank/service/strategy/UserService.java (hunk 4 above)
Comment (ghost) [reply src/main/java/org/growify/bank/service/strategy/UserService.java:35]: Great observation. In fact, the ModelMapper wasn’t configured yet during the development of the "UserService.java" class. I’ll make sure it’s properly injected.

File: src/main/java/org/growify/bank/service/strategy/UserService.java (hunk 5 above)
Comment (ghost) [reply src/main/java/org/growify/bank/service/strategy/UserService.java:37]: Thank you for the correction. Indeed, ResponseEntity should not be in the service layer. I’ll make the adjustment.

**Comment by ghost** [comment]
> Ensure the instructions are followed, then proceed with a new PR.

I’ll delve deeper into these points and submit the changes as soon as possible. Thank you very much for the review. If you have more suggestions, please let me know.

**Pull request closed without merging**